src/utils.cpp

```diff
@@ -195,6 +199,128 @@ bool is_lit(const Term & l, const Sort & boolsort)
   return false;
 }
 
+void cnf_to_dimacs(Term cnf, std::ostringstream & y)
+{
+  if (cnf->is_value() && cnf->to_string() == "true")
```

Comment:
Please also check if `cnf` is of Boolean sort. Otherwise, an exception should be thrown.